application/controllers/AdminController.php

```diff
@@ -498,6 +498,14 @@ public function getAdminFooter($url, $explanation, $return = false)
             $aData['versiontitle'] = gT('Version');
         }
 
+        $oEvent = new PluginEvent('beforeFooterRender');
+        App()->getPluginManager()->dispatchEvent($oEvent);
+        $data = $oEvent->get('data');
+
+        foreach ($data as $key => $item) {
```

Comment:
Isn't this just `array_merge`?

Reply:
Usage of `$event->append(` for plugins 

See https://manual.limesurvey.org/BeforeToolsMenuRender 

https://github.com/LimeSurvey/LimeSurvey/blob/26cd37a09c7250b75ef3e1a8f212ba8fb4e8be31/application/core/SurveyCommonAction.php#L774

**Warning** : broke if data is null with php8.1. (remind to activate debug for fix and new feature).

Reply:
To include custom data from database to javascript, one can use asset manager, register script, and the script can contain a global JS object with the custom data

```
App()->registerScript('name_of_script', <<<JAVASCRIPT
var installationData = {
  current_plan: $currentPlan  // Etc more data here
};
```